monai/transforms/utils.py

```diff
@@ -2051,5 +2058,46 @@ def has_status_keys(data: torch.Tensor, status_key: Any, default_message: str =
     return True, None
 
 
+def distance_transform_edt(
+    img: NdarrayOrTensor, sampling: None | float | list[float] = None, force_scipy: bool = False
+) -> NdarrayOrTensor:
+    """
+    Euclidean distance transform, either GPU based with CuPy / cuCIM
+    or CPU based with scipy.ndimage.
+    Choice depends on cuCIM being available or scipy can be forced with the ``force_scipy`` flag.
+
+    Note that the runtime running on the CPU may be really depending on the inputs size.
+
+    Args:
+        img: Input image on which the distance transform shall be run.
+            For CuPy shape must be ([channel, ]spatial_dim1, spatial_dim2 [, spatial_dim3]), so 2D, 3D and 4D input only.
+            If you need to run the transform on other shapes, use the ``force_scipy`` flag.
+            4D input gets passed channel-wise to cupy.
+        sampling: Spacing of elements along each dimension. If a sequence, must be of length equal to the input rank;
+            if a single number, this is used for all axes. If not specified, a grid spacing of unity is implied.
+        force_scipy: Force the CPU based scipy implementation of the euclidean distance transform
+    """
+    distance_transform_edt, has_cucim = optional_import(
+        "cucim.core.operations.morphology", name="distance_transform_edt"
+    )
+
+    if has_cp and has_cucim and not force_scipy:
```

Comment:
When we implement non-spatial dimensions and we care about performance here, I think we should use the option to pass in a preallocated torch tensor/numpyarray with the final dimension using the [distances and indices parameter](https://docs.rapids.ai/api/cucim/nightly/api/#cucim.core.operations.morphology.distance_transform_edt) (applies to [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.distance_transform_edt.html#scipy-ndimage-distance-transform-edt) as well). We then also need to make sure to match the proper datatypes. Keep in mind that we should support the float64_distances parameter for cucim here as well, since the float32 should be faster on GPU and the precision is probably not necessary in most cases. I am not 100% sure, but I think in case of cucim we should allocate a torch tensor and then use convert_to_cupy() to create a temporary view on this tensor cupy to pass to the transform. This way the result should get allocated using the pytorch memory pool